### ISSUE TITLE
Man pages aren't available by default

### ISSUE THREAD

**Issue by tomjakubowski** [brson/multirust#24]
At least after I installed multirust I can't seem to open the compiler's man page with `man rustc`. Instead I have to use an incantation like `man -l `multirust ctl toolchain-sysroot nightly`/share/man/man1/rustc.1`

Ideally I shouldn't be forced to update `$MANPATH` whenever I switch between toolchains.

**Comment by brson** [comment]
That's an interesting and unfortunate discovery. The best immediate solution might be to just have multirust install the man pages itself. They wouldn't match the toolchain man pages of course.

**Comment by alexreg** [comment]
Indeed, I discovered this too. It seems like the best solution would be to symlink the default toolchain man pages into `/usr/local/share/man` where appropriate. I presume this is what you already do with the binaries? (Then when the default toolchain is changed, update the symlinks accordingly.)

**Comment by tomjakubowski** [comment]
The `rustc`, `rustdoc`, `cargo`, etc. executables used in a multirust install are shell scripts I believe.

**Comment by alexreg** [comment]
Ah, I see. I wonder if there is a particular reason for this rather than just symlinking to the appropriate binaries? (I've only just installed multirust, so don't know much about it.) In any case, the man pages for default will probably need to be symlinked. Or at least the MANPATH env var will have to be changed according to the default toolchain.

**Comment by tomjakubowski** [comment]
Symlinks won't work because multirust supports using different toolchains at once from different shells, I believe.

**Comment by alexreg** [comment]
Fair enough, makes sense. Anyway, since the man pages would just correspond to the default toolchain, symlinking is maybe the best idea. Or altering MANPATH perhaps?

Perhaps man pages for multiple toolchains could be made accessible even, so that `man rustc` views the manual for the default rustc, but `man rustc-stable` does the same for the ‘stable' toolchain.

**Comment by tomjakubowski** [comment]
I don't see anything wrong with multirust changing `MANPATH` when switching toolchains, but this would unfortunately mean suggesting that some rbenv-esque incantation like `eval $(multirust init -)` be added to user .shellrc files.

**Comment by alexreg** [comment]
That’s true. But symlinking _just_ the man pages (for default and other toolchains) sounds like a fair enough alternative too. The project author  will hopefully choose one of these suggestions, at least.

**Comment by brson** [comment]
Overriding manpath would require adding a command like @tomjakubowski says that creates a new shell with the correct environment. It's pretty ugly, and people would have to know 'if i want to read the man page put me in the correct environment'.

I _do_ think we'll need such functionality to make multirust work correctly on windows where there's no RPATH.

Another option might be to install dummy manpages that just say 'run multirust man <foo>', then add manpage display to multirust.

**Comment by alexreg** [comment]
Hi. Any update on this now? What's the current recommended way to access the man pages?

**Comment by brson** [comment]
@alexreg No update. My current preference is to add a `multirust man` command, but I haven't thought about it lately.

**Comment by alexreg** [comment]
Okay. I agree that would be a relatively good solution… I’ll surely get by, until you implement it.

**Comment by steveklabnik** [comment]
:+1: to having some way to get man pages installed

**Comment by brson** [comment]
In addition to adding `multirust man`, multirust should probably install some placeholder man pages that say 'run multirust man'.

**Comment by alexreg** [comment]
Any update on this?

**Comment by brson** [comment]
@alexreg No, I am not intending to work on it myself, but I will accept a patch that adds `multirust man` along with some placeholder man pages.

**Comment by alexreg** [comment]
Okay. Alternatively, is it coming in multirust-rc a.k.a. rustup?

**Comment by brson** [comment]
@alexreg It will probably happen eventually. [Here's an isue for it](https://github.com/rust-lang-nursery/multirust-rs/issues/270). It's a difficult question since man pages don't exist on windows and I want to maintain feature parity on windows. To implement it in rustup we might want our own troff interpreter.

**Comment by alexreg** [comment]
Fair enough. Maybe just direct Windows users to online versions of the man pages? :)

Windohs is a bit special needs anyway, as they say...
